src/counter_op.rs

```diff
@@ -15,19 +16,21 @@ impl OpcodeCounterInspector {
     }
 
     /// Returns the opcode counts collected during transaction execution.
-    pub fn opcode_counts(&self) -> &HashMap<u8, u64> {
+    pub fn opcode_counts(&self) -> &HashMap<OpCode, u64> {
         &self.opcode_counts
     }
 }
 
-impl<DB> Inspector<DB> for OpcodeCounterInspector
+impl<DB> revm::Inspector<DB> for OpcodeCounterInspector
 where
-    DB: Database,
+    DB: revm::Database,
 {
-    fn step(&mut self, interp: &mut Interpreter, _context: &mut EvmContext<DB>) {
+    fn step(&mut self, interp: &mut Interpreter, _context: &mut revm::EvmContext<DB>) {
         let opcode_value = interp.current_opcode();
+        let opcode = OpCode::new(opcode_value)
+            .unwrap_or_else(|| unsafe { OpCode::new_unchecked(opcode_value) });
```

Comment:
it's possible that this is an invalid opcode, so we can't use unsafe here and instead need to try and only if it's a valid opcode add it to the set

Reply:
> it's possible that this is an invalid opcode, so we can't use unsafe here and instead need to try and only if it's a valid opcode add it to the set

Thanks for your review Mat! Changed this :)

Reply:
if you don't mind i need some help to count the gas of the opcode, since [sslivkoff] doesn't want the multiplication i dunno how i would track this

Reply:
cc @mattsse  its possible to have review of this pls

Reply:
thanks, almost done